ddl/reorg.go

```diff
@@ -213,6 +213,7 @@ func (w *worker) runReorgJob(rh *reorgHandler, reorgInfo *reorgInfo, tblInfo *mo
 		if job.IsCancelling() {
 			return dbterror.ErrCancelledDDLJob
 		}
+		tryToUpdateRealReorgRange(rh, reorgInfo, tbl)
```

Comment:
Could we move the logic of [line635](https://github.com/pingcap/tidb/pull/38195/files#diff-51926b25b437247805c6880b2fd041f096575b30050b560ab934f1d863bb6bf6R635)-[line678](https://github.com/pingcap/tidb/pull/38195/files#diff-51926b25b437247805c6880b2fd041f096575b30050b560ab934f1d863bb6bf6R678) here? Then we needn't do twice this logic.

Reply:
Then we have to change `reorgInfo.first` related code. Let's keep it in this PR.